src/extension/promptService/promptService.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Configstore from "configstore";
```

Comment:
While this is enough for our current scenario, it might not scale well in the (near) future.

My idea is that we could have multiple experiments running at the same time, and that not all of those will prompt the user for permission. For example, if we could have both js debuggers running at the same time I'd rather use this approach than asking the user to switch to another extension.
This could be the same for other features that we would like to test more thoroughly before enabling for the whole population.

What do you think of having an `experimentService.ts` that downloads the JS and controls the whole status of the experiments for the given user: if it has to be active, deleted because it disappeared from the config, etc.
Then we would have experiments query this service to know their status and enable their features or not. In this particular case it would be prompt the user (only once), but with my previous example of debuggers it would choose one or the other.

Also I think we should send some telemetry when a given experiment is activated.

Does it make sense? Happy to jump on a call if it would make things easier :)

Reply:
@molant, Do I understand you correctly that the idea is that the code of experiments will be located on a remote server and we should just download and execute it then on the extension side?

Reply:
> the code of experiments will be located on a remote server 

No, the code will be in the extension. The `experimentService.ts` will be in charge of doing the network request to know what experiments we currently have (same as what `promptService.ts` does in this code).
Let me know if you want to jump on a call!